src/pybind/mgr/dashboard/controllers/cephfs.py

```diff
@@ -639,6 +639,17 @@ def rm_snapshot(self, fs_id, path, name):
         cfs = self._cephfs_instance(fs_id)
         cfs.rm_snapshot(path, name)
 
+    @RESTController.Resource('PUT', path='/rename-path')
+    def rename_path(self, fs_id, src_path, dst_path) -> None:
```

Comment:
Just saw this PR (CC @nizamial09). The CephFS component is not the best implemented according to REST guidelines, but this should be something like:

```
PATCH /api/cephfs/{fsid}/tree/{path}
  Request body:
  {
    "path": {new_path}
  }
```
`PUT` implies idempotency, and renaming is obviously not idempotent. And we shouldn't include verbs in endpoints, just nouns (`cephfs`, `tree`).

Reply:
Hi Ernesto,
We have discussed this issue before, `tree` usually represents directory operations, `rename` can also use to file. So we decided not to implement it in the `tree`

Reply:
thanks @epuertat, i was a bit confused as well with the naming convention here and I still don't know with the example you provided, how you can sent the current path in the req.

`PATCH /api/cephfs/{fsid}/tree/random/path/I/want/to/rename` or will it be something else? how will that `/` gets processed.